src/api/app/models/user.rb

```diff
@@ -840,6 +840,15 @@ def mark_login!
     update_attributes(last_logged_in_at: Time.now, login_failure_count: 0)
   end
 
+  def pending_announcement
```

Comment:
This is a [scope](https://guides.rubyonrails.org/active_record_querying.html#passing-in-arguments) not a method.

Reply:
@hennevogel How would you do it?

Reply:
So something like:

```ruby
def self.pending_for(user)
  return if user.announcements.include?(Announcement.last)
  Announcement.last
end
```
?

I don't see any real difference here. Beside it's the user that has to take an action on the anouncement, not the other way around.